src/scaler/spreadscaler/link.rs

```diff
@@ -122,209 +122,145 @@ where
-            //             link_name: self.config.provider_link_name.to_owned(),
-            //             model_name: self.config.model_name.to_owned(),
-            //         })]
-            //     })
-            //     .unwrap_or_default();
-
-            // if exists && !values_different {
-            //     trace!("Linkdef already exists, skipping");
-            // } else if !exists || values_different {
-            //     trace!("Linkdef does not exist or needs to be recreated");
-            //     commands.push(Command::PutLinkdef(PutLinkdef {
-            //         actor_id: actor_id.to_owned(),
-            //         provider_id: provider_id.to_owned(),
-            //         link_name: self.config.provider_link_name.to_owned(),
-            //         contract_id: self.config.provider_contract_id.to_owned(),
-            //         values: self.config.values.to_owned(),
-            //         model_name: self.config.model_name.to_owned(),
-            //     }))
-            // };
-
-            let commands = if !exists {
-                *self.status.write().await = StatusInfo::reconciling(&format!(
-                    "Putting link definition between {actor_id} and {provider_id}"
-                ));
-                vec![Command::PutLinkdef(PutLinkdef {
-                    actor_id: actor_id.to_owned(),
-                    provider_id: provider_id.to_owned(),
-                    link_name: self.config.provider_link_name.to_owned(),
-                    contract_id: self.config.provider_contract_id.to_owned(),
-                    values: self.config.values.to_owned(),
-                    model_name: self.config.model_name.to_owned(),
-                })]
-            } else {
-                *self.status.write().await = StatusInfo::deployed("");
-                Vec::with_capacity(0)
-            };
-            Ok(commands)
-        } else {
-            trace!("Actor ID and provider ID are not initialized, skipping linkdef creation");
+        let source_id = &self.config.source_id;
+        let target = &self.config.target;
+        let linkdefs = self.ctl_client.get_links().await?;
+        let (exists, _config_different) = linkdefs
+            .into_iter()
+            .find(|linkdef| {
+                &linkdef.source_id == source_id
+                    && &linkdef.target == target
+                    && linkdef.name == self.config.name
+            })
+            .map(|linkdef| {
+                (
+                    true,
+                    linkdef.source_config != self.config.source_config
+                        || linkdef.target_config != self.config.target_config,
+                )
+            })
+            .unwrap_or((false, false));
+
+        // TODO(brooksmtownsend): Now that links are ID based not public key based, we should be able to reenable this
+        // TODO: Reenable this functionality once we figure out https://github.com/wasmCloud/wadm/issues/123
```

Comment:
As per my other comment, I think we can reenable this

Reply:
I did it